mscore/scoretreewidget.cpp

```diff
@@ -0,0 +1,51 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+#include "scoretreewidget.h"
+
+#include "libmscore/score.h"
+#include "scoreitemmodel.h"
+#include "scoretreeview.h"
+
+namespace Ms {
+//---------------------------------------------------------
+//   ScoreTreeWidget
+//---------------------------------------------------------
+
+ScoreTreeWidget::ScoreTreeWidget(QMainWindow* parent)
+    : QDockWidget("Score Tree", parent)
+{
+    _treeView = new ScoreTreeView(this);
+    setWidget(_treeView);
+    setAllowedAreas(Qt::RightDockWidgetArea);
+    parent->addDockWidget(Qt::RightDockWidgetArea, this);
+}
+
+//---------------------------------------------------------
+//   ~ScoreTreeWidget
+//---------------------------------------------------------
+
+ScoreTreeWidget::~ScoreTreeWidget()
+{
+}
+
+//---------------------------------------------------------
+//   setScore
+//---------------------------------------------------------
+
+void ScoreTreeWidget::setScore(Score* s)
+{
+    QAbstractItemModel* oldModel = _treeView->model();
+    _treeView->setModel(new ScoreItemModel(s, _treeView));
```

Comment:
Must check for `nullptr` before creating a new model otherwise it leads to a crash when there are no scores open:

```c++
if (s) {
    _treeView->setModel(new ScoreItemModel(s, _treeView));
}
```

Keep the code to delete the existing model as this will clear the tree view, which is good if there are no scores open.